src/UnitTests.Shared/TestEnvironment.cs

```diff
@@ -83,6 +87,8 @@ private void Cleanup()
             {
                 _disposed = true;
 
+                // It should be in reverse order revert to get back to original state.
+                _variants.Reverse();
                 // Reset test variants
                 foreach (var variant in _variants)
                 {
```

Comment:
Curious, is this a pro-active fix or did you actually run into variants that depend on the order?

also, nit:

```suggestion
                // Reset test variants in reverse order to get back to original state.
                for (int i = _variants.Count - 1; i >= 0; i--)
                {
                    var variant = _variants[i];
```

Reply:
While fixing the issue I run into the problem from the order resetting variants.
Given that with `SET MSBUILDLIVELOGGER=auto` we run tests https://github.com/dotnet/msbuild/blob/6fec6072f0591bede57a71474ddc08b95e3e4c28/src/Build.UnitTests/TerminalLoggerConfiguration_Tests.cs#L139-L142, we need to first clear pre-setting `MSBUILDLIVELOGGER`, then set the value for the test. For the variants keeping the value for environment variable `MSBUILDLIVELOGGER`, they are added by the following order.
||original value|current value|
|-|-|-|
|clear pre-setting|auto|null|
|set the value for the test|null|true|

It should revert the variants in reverse order to get back to original value `auto`.

Reply:
Makes perfect sense, thank you!